typescript/src/lib/utils/ledger.ts

```diff
@@ -0,0 +1,177 @@
+    walletApiClient?: WalletAPIClient
+  ) {
+    super()
+    ethers.utils.defineReadOnly(this, "provider", provider || null)
+    this._windowMessageTransport =
+      windowMessageTransport || getWindowMessageTransport()
+    this._walletApiClient =
+      walletApiClient || getWalletAPIClient(this._windowMessageTransport)
+  }
+
+  get account() {
+    return this._account
+  }
+
+  setAccount(account: Account | undefined): void {
+    this._account = account
+  }
+
+  async requestAccount(
+    params: { currencyIds?: string[] | undefined } | undefined
+  ): Promise<Account> {
+    this._windowMessageTransport.connect()
+    const account = await this._walletApiClient.account.request(params)
+    this._windowMessageTransport.disconnect()
+    this._account = account
+    return this._account
+  }
+
+  getAccountId(): string {
+    if (!this._account || !this._account.id) {
+      throw new AccountNotFoundError()
+    }
+    return this._account.id
+  }
+
+  async getAddress(): Promise<string> {
+    if (!this._account || !this._account.address) {
+      throw new AccountNotFoundError()
+    }
+    return this._account.address
+  }
+
+  async signMessage(message: string): Promise<string> {
+    if (!this._account || !this._account.address) {
+      throw new AccountNotFoundError()
+    }
+    this._windowMessageTransport.connect()
+    const buffer = await this._walletApiClient.message.sign(
+      this._account.id,
+      Buffer.from(message)
+    )
+    this._windowMessageTransport.disconnect()
+    return buffer.toString()
+  }
+
+  async signTransaction(
+    transaction: ethers.providers.TransactionRequest
+  ): Promise<string> {
+    if (!this._account || !this._account.address) {
+      throw new AccountNotFoundError()
+    }
+
+    const { value, to, nonce, data, gasPrice, gasLimit } = transaction
+
+    const ethereumTransaction: any = {
+      family: "ethereum" as const,
+      amount: value ? new BigNumber(value.toString()) : new BigNumber(0),
+      recipient: to ? to : AddressZero,
+    }
+
+    if (nonce) ethereumTransaction.nonce = nonce
+    if (data)
+      ethereumTransaction.data = Buffer.from(
+        Hex.from(data.toString()).toString(),
+        "hex"
+      )
+    if (gasPrice)
+      ethereumTransaction.gasPrice = new BigNumber(gasPrice.toString())
+    if (gasLimit)
+      ethereumTransaction.gasLimit = new BigNumber(gasLimit.toString())
+
+    this._windowMessageTransport.connect()
+    const buffer = await this._walletApiClient.transaction.sign(
+      this._account.id,
+      ethereumTransaction
+    )
+    this._windowMessageTransport.disconnect()
+    return buffer.toString()
+  }
+
+  async sendTransaction(
+    transaction: Deferrable<ethers.providers.TransactionRequest>
+  ): Promise<ethers.providers.TransactionResponse> {
+    if (!this._account || !this._account.address) {
+      throw new AccountNotFoundError()
+    }
+
+    const { value, to, nonce, data, gasPrice, gasLimit } = transaction
```

Comment:
Are we sure this will work in all cases? `Deferrable` is not clear to me.

Reply:
It should work now. See description of 54be6d7c7e86b6a13a81ac71a74255bc23b05b9d